vmdb/spec/models/ems_refresh/refreshers/foreman_refresher_spec.rb

```diff
@@ -0,0 +1,113 @@
+require "spec_helper"
+
+describe EmsRefresh::Refreshers::ForemanRefresher do
+  # where clause to use spec related entries
+  let(:spec_related) { "name like 'ProviderRefreshSpec%'" }
+  let(:provider) do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    FactoryGirl.create(:provider_foreman,
+                       :zone       => zone,
+                       :url        => "10.8.96.102",
+                       :verify_ssl => false,
+                      )
+  end
+
+  it "will perform a full refresh on api v2" do
+    VCR.use_cassette("#{described_class.name.underscore}_api_v2") do
+      EmsRefresh.refresh(provider)
+    end
+
+    assert_table_counts
+    assert_provisioning_manager
+    assert_configuration_manager
+  end
+
+  def assert_table_counts
+    expect(Provider.count).to                   eq(1)
+    expect(CustomizationScript.count).to        eq(19)
+    expect(CustomizationScriptPtable.count).to  eq(11)
+    expect(CustomizationScriptMedium.count).to  eq(8)
+    expect(OperatingSystemFlavor.count).to      eq(5)
+    expect(ConfigurationProfile.count).to       eq(14)
+    expect(ConfiguredSystem.count).to           eq(39)
+  end
+
+  def assert_provisioning_manager
+    manager = Provider.first.provisioning_manager
+    expect(manager).not_to be_nil
+
+    css = manager.customization_scripts
+    expect(css.count).to                              eq(19)
+    expect(css.ptables.count).to                      eq(11)
+    expect(css.media.count).to                        eq(8)
+    expect(manager.operating_system_flavors.count).to eq(5)
+
+    medium = css.media.where(spec_related).first
+    ptable = css.ptables.where(spec_related).first
+    osf = manager.operating_system_flavors.where(spec_related).first
+
+    expect(medium.attributes).to include(
+      "name"        => "ProviderRefreshSpec-Media",
+      "type"        => "CustomizationScriptMedium",
+      "manager_ref" => "medium:8"
+    )
```

Comment:
We have a special matcher for this which you might have seen from the other specs...this can be:

```ruby
expect(medium).to have_attributes(
  :name        => "ProviderRefreshSpec-Media",
  :type        => "CustomizationScriptMedium",
  :manager_ref => "medium:8"
)